x-pack/platform/plugins/shared/automatic_import_v2/server/services/samples_index/storage.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import type { IndexStorageSettings } from '@kbn/storage-adapter';
+import { StorageIndexAdapter, types } from '@kbn/storage-adapter';
+
+export const automaticImportSamplesIndexName = 'automatic-import-samples';
+
+const automaticImportSamplesIndexAdapterSettings = {
+  name: automaticImportSamplesIndexName,
+  schema: {
+    properties: {
+      // ONLY map fields we actively search/filter/aggregate on
+      integration_id: types.keyword({ ignore_above: 256 }),
```

Comment:
Might be too complex at this point but what do you think of refactoring out schemas so it is easier to find later in the folder structure.